File: BankingApp/BankBackend/DataAccess/Entities/Transaction.cs
using System;
using System.Collections.Generic;

namespace DataAccess.Entities;

public partial class Transaction
{
    public int Id { get; set; }

    public decimal? Amount { get; set; }

    public int? CardId { get; set; }

    public int? AccountId { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? Description { get; set; }

    public int? RecipientId { get; set; }

    public int? Status { get; set; }

    public int? SenderId { get; set; }

    public bool? RecpientType { get; set; }

    public bool? SenderType { get; set; }

    // public virtual User? Recipient { get; set; }

    // public virtual User? Sender { get; set; }
}

File: BankingApp/BankFrontend/FrontEnd/src/app/models/transaction.ts
export interface Transaction {
    "id"?: number,
    "amount"?: number,
    "cardId"?: number,
    "accountId"?: number,
    "createdAt"?: string | Date,
    "description"?: string,
    "recipientId"?: number,
    "status"?: number,
    "senderId"?: number,
    "recpientType"?: boolean,
    "senderType"?: boolean
}